src/SnD.Sdk.Storage.Minio/MinioService.cs

```diff
@@ -30,12 +31,34 @@ public MinioService(IMinioClient minioClient)
     /// <param name="bucketName">The name of the bucket where the object is stored.</param>
     /// <param name="objectName">The name of the object to retrieve.</param>
     /// <returns>A Task representing the asynchronous operation.</returns>
-    public Task GetObjectAsync(string bucketName, string objectName)
+    public Task<ObjectStat> GetObjectAsync(string bucketName, string objectName)
     {
         GetObjectArgs args = new GetObjectArgs().WithBucket(bucketName).WithObject(objectName);
         return minioClient.GetObjectAsync(args);
     }
 
+
+    /// <summary>
+    /// Asynchronously reads an object from a specified bucket and returns its content as a <see cref="Stream"/>.
+    /// </summary>
+    /// <param name="bucketName">The name of the bucket where the object is stored.</param>
+    /// <param name="objectName">The name of the object to read.</param>
+    /// <returns>A <see cref="Task"/> representing the asynchronous operation, which upon completion returns a <see cref="Stream"/> containing the object's content.</returns>
+    public async Task<Stream> ReadObjectAsync(string bucketName, string objectName)
+    {
+        var memoryStream = new MemoryStream();
+        await minioClient.GetObjectAsync(new GetObjectArgs()
+            .WithBucket(bucketName)
+            .WithObject(objectName)
+            .WithCallbackStream(stream =>
```

Comment:
Is this `CopyTo` operation eager or lazy? Maybe it would be easier to convert the result `stream` object into a byte array and return the array?

Reply:
It was eager, changed it to `CopyToAsync`. I would like to keep it as a `Stream` because it's needed as input for the ParquetReader in the Parquet .NET library.